LibSWBF2/API.cpp

```diff
@@ -299,6 +322,77 @@ namespace LibSWBF2
     }
 
 
+    //CollisionPrimitive
+
+    const void CollisionPrimitive_FetchAllFields(CollisionPrimitive *primPtr,
+                                            float_t& f1, float_t& f2, float_t& f3,
+                                            const char *& namePtr, const char *& parentNamePtr,
+                                            uint32_t& maskFlags, uint32_t& primitiveType,
+                                            Vector3*& pos, Vector4*& rot)
+    {
+    	static String name, parentName;
+    	static Vector3 posTemp;
+    	static Vector4 rotTemp;
+
+    	f1 = f2 = f3 = 0.0f;
+
+    	name = primPtr -> GetName();
+    	parentName = primPtr -> GetParentName();
+
+    	namePtr = name.Buffer();
+    	parentNamePtr = parentName.Buffer();
+
+    	maskFlags = (uint32_t) primPtr -> GetMaskFlags();
+    	primitiveType = (uint32_t) primPtr -> GetPrimitiveType();
+
+    	rotTemp = primPtr -> GetRotation();
+    	posTemp = primPtr -> GetPosition();
+
+    	pos = &posTemp; 
+    	rot = &rotTemp; 
+
+    	switch (primitiveType)
+    	{
+    		case 1:
+    			primPtr -> GetSphereRadius(f1);
+    			return;
+    		case 2:
+    			primPtr -> GetCylinderDims(f1,f2);
+    			return;
+    		case 4:
+    			primPtr -> GetCubeDims(f1,f2,f3);
+    			return;
+    		default:
+    			return;
+    	}
+    }
+
+
+
+    const void Vector4_FromPtr(const Vector4* vec, float& x, float& y, float& z, float &w)
+    {
```

Comment:
Writing these wrappers is tedious as you know, so for classes that don't hold that much data I just write Fetch methods that get all fields at once.